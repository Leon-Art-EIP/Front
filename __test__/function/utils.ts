import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
