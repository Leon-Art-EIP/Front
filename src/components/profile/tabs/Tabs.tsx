export interface ITabsProps {
  publicationsTabOnClick: () => void;
  collectionsTabOnClick: () => void;
  aboutTabOnClick: () => void;
}

export default function Tabs(props: ITabsProps): JSX.Element {
  return (
    <div className="bg-[#ECECEC] inline-flex gap-2 rounded-full px-1">
      <button
        className="flex-1 text-center rounded-full border-r-2 border-white py-2 pl-3 pr-5"
        onClick={props.publicationsTabOnClick}
      >
        Publications
      </button>
      <button
        className="flex-1 text-center rounded-full border-r-2 border-white py-2 pl-3 pr-5"
        onClick={props.collectionsTabOnClick}
      >
        Collections
      </button>
      <button className="flex-1 text-center rounded-full border-r-2 py-2 pl-3 pr-5" onClick={props.aboutTabOnClick}>
        Ã€ propos
      </button>
    </div>
  );
}
