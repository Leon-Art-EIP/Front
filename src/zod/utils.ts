export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
