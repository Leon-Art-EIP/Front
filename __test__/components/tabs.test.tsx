import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Tabs from "../../src/components/profile/tabs/Tabs";

test("Tabs about snapshot", () => {
  testSnapshot(
    <Tabs
      aboutTabOnClick={() => {}}
      collectionsTabOnClick={() => {}}
      publicationsTabOnClick={() => {}}
      selectedTab="about"
    />
  );
});

test("Tabs collections snapshot", () => {
  testSnapshot(
    <Tabs
      aboutTabOnClick={() => {}}
      collectionsTabOnClick={() => {}}
      publicationsTabOnClick={() => {}}
      selectedTab="collections"
    />
  );
});

test("Tabs publications snapshot", () => {
  testSnapshot(
    <Tabs
      aboutTabOnClick={() => {}}
      collectionsTabOnClick={() => {}}
      publicationsTabOnClick={() => {}}
      selectedTab="publications"
    />
  );
});
