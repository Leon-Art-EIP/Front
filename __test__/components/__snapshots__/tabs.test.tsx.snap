// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs about snapshot 1`] = `
<div
  className="relative h-[40px] rounded-full"
>
  <button
    className="absolute w-1/3 h-full z-20 grid grid-cols-1 items-center border-4 border-white rounded-full text-center truncate bg-background-hl"
    onClick={[Function]}
  >
    Publications
  </button>
  <button
    className="absolute w-2/3 h-full z-10 flex items-center border-4 border-white rounded-full bg-background-hl"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      Collections
    </div>
  </button>
  <button
    className="absolute w-full h-full z-0 flex items-center border-4 border-white rounded-full bg-primary text-white"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      A propos
    </div>
  </button>
</div>
`;

exports[`Tabs collections snapshot 1`] = `
<div
  className="relative h-[40px] rounded-full"
>
  <button
    className="absolute w-1/3 h-full z-20 grid grid-cols-1 items-center border-4 border-white rounded-full text-center truncate bg-background-hl"
    onClick={[Function]}
  >
    Publications
  </button>
  <button
    className="absolute w-2/3 h-full z-10 flex items-center border-4 border-white rounded-full bg-primary text-white"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      Collections
    </div>
  </button>
  <button
    className="absolute w-full h-full z-0 flex items-center border-4 border-white rounded-full bg-background-hl"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      A propos
    </div>
  </button>
</div>
`;

exports[`Tabs publications snapshot 1`] = `
<div
  className="relative h-[40px] rounded-full"
>
  <button
    className="absolute w-1/3 h-full z-20 grid grid-cols-1 items-center border-4 border-white rounded-full text-center truncate bg-primary text-white"
    onClick={[Function]}
  >
    Publications
  </button>
  <button
    className="absolute w-2/3 h-full z-10 flex items-center border-4 border-white rounded-full bg-background-hl"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      Collections
    </div>
  </button>
  <button
    className="absolute w-full h-full z-0 flex items-center border-4 border-white rounded-full bg-background-hl"
    onClick={[Function]}
  >
    <div
      className="flex-1"
    />
    <div
      className="flex-1"
    />
    <div
      className="flex-1 text-center truncate"
    >
      A propos
    </div>
  </button>
</div>
`;
